autoPyTorch/evaluation/train_evaluator.py

```diff
@@ -79,10 +79,23 @@ class TrainEvaluator(AbstractEvaluator):
             An optional dictionary to include components of the pipeline steps.
         exclude (Optional[Dict[str, Any]]):
             An optional dictionary to exclude components of the pipeline steps.
-        disable_file_output (Union[bool, List[str]]):
-            By default, the model, it's predictions and other metadata is stored on disk
-            for each finished configuration. This argument allows the user to skip
-            saving certain file type, for example the model, from being written to disk.
+        disable_file_output (Optional[List]):
+                Used as a list to pass more fine-grained
+                information on what to save. Allowed elements in the list are:
+
+                + `y_optimization`:
+                    do not save the predictions for the optimization set,
+                    which would later on be used to build an ensemble. Note that SMAC
+                    optimizes a metric evaluated on the optimization set.
+                + `pipeline`:
+                    do not save any individual pipeline files
+                + `pipelines`:
```

Comment:
So currently `pipeline` is being used to save the model for holdout resampling and `pipelines` is being used to save the model for cross-validation. I think we can just have `pipeline` instead of the two. Actually, we had an unnoticed logical error before when using cross-validation. Where we were storing a pipeline that has not been fitted yet. In my upcoming PR, I have removed this functionality as it was confusing and unnecessary. So, I'll remove `pipelines` from the set of options in that PR.